operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcherTest.java

```diff
@@ -56,9 +58,12 @@ void setup() {
     when(configuration.getName()).thenReturn("EventDispatcherTestController");
     when(configService.getMetrics()).thenReturn(Metrics.NOOP);
     when(configuration.getConfigurationService()).thenReturn(configService);
-    when(configService.getResourceCloner()).thenReturn(ConfigurationService.DEFAULT_CLONER);
-    when(reconciler.reconcile(eq(customResource), any()))
-        .thenReturn(UpdateControl.updateResource(customResource));
+    when(configService.getResourceCloner()).thenReturn(new Cloner() {
```

Comment:
Why was this changed?

Reply:
To make sure to make easier to do `when(..).thenReturn()`  with mockito. Since otherwise its a different instance.

Reply:
Yes, but then you're not really testing the actual behavior?

Reply:
This can be understood as a form of a mocking of that clone method / cloner. The goal here is not to test the cloner but only if it is called in case it's relevant.

Reply:
I understand but the problem is what if the behavior changes when the instance is not actually cloned?

Reply:
Ahh, I think I see what you mean, will try to improve it.

Reply:
Actually not, is just what instance we move around, for some special cases.